contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,494 @@
+  name: local-fast
+provisioner: kubernetes.io/local-block-ssd
+parameters:
+  volumeType: block
+```
+
+***This has implementation details that have yet to be determined. It is included in this proposal for completeness of design ****
+
+## UC6: 
+
+DESCRIPTION: The developer wishes to request a block device via a Storage Class.
+
+WORKFLOW:
+
+USER: 
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+ name: pvc-local-block
+spec:
+ volumeType: block
+ storageClassName: local-fast
+ accessModes:
+  - ReadWriteOnce
+ resources:
+   requests:
+     storage: 10Gi
+```  
+
+## UC7: 
+
+DESCRIPTION: Admin creates network raw block devices
+
+BACKGROUND: Admin wishes to pre-create Persistent Volumes in GCE as raw block devices
+
+WORKFLOW:
+
+ADMIN:
+
+```
+apiVersion: "v1"
+kind: "PersistentVolume"
+metadata:
+  name: gce-disk-1
+  annotations:
+    volume.beta.kubernetes.io/mount-options: "discard"
+Spec:
+  volumeType: block
+  capacity:
+    storage: "10Gi"
+  accessModes:
+    - "ReadWriteOnce"
+  gcePersistentDisk:
+    fsType: "block"
+    pdName: "gce-disk-1"
+```
+
+***If admin specifies volumeType: block + fstype: ext4 then they would have the default behavior of files on block ***
+***fsType values will be provisioner dependent. Block is suggested for development simplicity. Since the PVC object is passed
+   to the provisioner, it will be responsible for validating and handling whether or not it supports the volumeType being passed ***
+
+# Container Runtime considerations
+It is important the values that are passed to the container runtimes are valid and support the current implementation of these various runtimes. Listed below are a table of various runtime and the mapping of their values to what is passed from the mount. 
+
+| runtime engine    | runtime options  | accessMode       |  
+| --------------    |:----------------:| ----------------:|
+| docker/runc/rkt   |  mknod / RWM     | RWO              |
+| docker/runc/rkt   |       R          | ROX              |
+
+The accessModes would be passed as part of the options array and would need validate against the specific runtime engine. 
+Since rkt doesn't use the CRI, the config values would need to be passed in the legacy method.
+
+The runtime option would be placed in the DeviceInfo as such:
+devices = append(devices, kubecontainer.DeviceInfo{PathOnHost: path, PathInContainer: path, Permissions: "mrw"}) for RWO
+
+The implemenation plan would be to rename the current makeDevices ot makeGPUDevices and create a seperate function to add the raw block devices to the option array to be passed to the container runtime. This would interate on the paths passed in for the pod/container.
+
+# Implementation Plan, Features & Milesones
+
+Phase 1: v1.8
+Feature: Pre-provisioned PVs to precreated devices 
+
+               Milestone 1: API changes
+
+               Milestone 2: Restricted Access 
+               
+               Milestone 3: Scrubbing device after PV is deleted
```

Comment:
This should be removed as it is specific to local storage and not the general volume layer.